x-pack/plugins/cloud_security_posture/server/routes/status/status.ts

```diff
@@ -74,16 +90,23 @@ const getCspStatus = async ({
   agentPolicyService,
   agentService,
 }: CspApiRequestHandlerContext): Promise<CspSetupStatus> => {
-  const [hasFindings, installation, latestCspPackage, installedPackagePolicies] = await Promise.all(
-    [
-      checkForFindings(esClient.asCurrentUser, true, logger),
-      packageService.asInternalUser.getInstallation(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
-      packageService.asInternalUser.fetchFindLatestPackage(CLOUD_SECURITY_POSTURE_PACKAGE_NAME),
-      getCspPackagePolicies(soClient, packagePolicyService, CLOUD_SECURITY_POSTURE_PACKAGE_NAME, {
-        per_page: 10000,
-      }),
-    ]
-  );
+  const [
+    findingsLatestIndexStatus,
+    findingsIndexStatus,
+    scoreIndexStatus,
+    installation,
+    latestCspPackage,
+    installedPackagePolicies,
+  ] = await Promise.all([
+    checkIndexStatus(esClient.asCurrentUser, LATEST_FINDINGS_INDEX_DEFAULT_NS, logger),
+    checkIndexStatus(esClient.asCurrentUser, FINDINGS_INDEX_PATTERN, logger),
+    checkIndexStatus(esClient.asCurrentUser, BENCHMARK_SCORE_INDEX_DEFAULT_NS, logger),
```

Comment:
note that this `Promise.all` effectively bundles all our "state computations". 
so for example, if a user doesn't have privileges, instead of notifying them immediately and stopping all work, we will first try compute all other relevant data pieces we need to figure out the state.  

using a single function that has all the data to compute the state (`calculateCspStatusCode`) is kinda wasteful as much of the states can be computed sepereatly, like permissions.

Reply:
consider extracting all the `checkIndexStatus` to its own `Promise.all` that fires before everything and if user doesn't have permissions, bail early without doing extra work.

Reply:
as an API I do think there's value in giving back the entire status, and not partial pieces to correspond to different cases.

In case it was an ACCESS_DENIED error I agree with you this should be returned asap